MapleServer2/PacketHandlers/Game/InsigniaHandler.cs

```diff
@@ -1,28 +1,69 @@
-﻿using MaplePacketLib2.Tools;
+﻿using System;
+using System.Collections.Generic;
+using Maple2Storage.Types;
+using MaplePacketLib2.Tools;
 using MapleServer2.Constants;
+using MapleServer2.Data.Static;
 using MapleServer2.Packets;
 using MapleServer2.Servers.Game;
+using MapleServer2.Types;
 using Microsoft.Extensions.Logging;
 
 namespace MapleServer2.PacketHandlers.Game
 {
     public class InsigniaHandler : GamePacketHandler
     {
-        public override RecvOp OpCode => RecvOp.NAMETAG_SYMBOL;
+        public override RecvOp OpCode => RecvOp.INSIGNIA;
 
         public InsigniaHandler(ILogger<InsigniaHandler> logger) : base(logger) { }
 
         public override void Handle(GameSession session, PacketReader packet)
         {
+
             short insigniaId = packet.ReadShort();
 
-            if (insigniaId < 0)
+            if (insigniaId < 0 && !InsigniaMetadataStorage.IsValid(insigniaId))
             {
                 return;
             }
 
+            if (!CanEquipInsignia(session, insigniaId))
+            {
+                insigniaId = 0;
+            }
+
```

Comment:
Should we be setting a user's insignia to 0? How it works in game is that you _are_ allowed to switch to an insignia you don't have but what matters is the following byte whether it should display or not depending if you have it.

Reply:
I don't know what is the best way, both are good for me. I guess I will change to: "if you don't have just doesn't show the insignia" since its the way nexon does it.